src/zenml/hooks/hook_validators.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Validation functions for hooks."""
+
+import inspect
+from types import FunctionType
+from typing import TYPE_CHECKING, Union
+
+from zenml.utils import source_utils
+
+if TYPE_CHECKING:
+    HookSpecification = Union[str, FunctionType]
+
+
+def resolve_and_validate_hook(hook_func: "HookSpecification") -> str:
+    """Resolves and validates a hook callback.
+
+    Args:
+        hook_func: Callable hook function.
+
+    Returns:
+        str: Resolved source path of `hook_func`.
+
+    Raises:
+        ValueError: If `hook_func` is not a valid callable.
+    """
+    if type(hook_func) is str:
+        func = source_utils.load_source_path(hook_func)
+    else:
+        # This means hook is a callable
+        func = hook_func
+
+    if not callable(func):
+        raise ValueError(f"{func} is not a valid function.")
+
+    from zenml.steps.base_parameters import BaseParameters
+    from zenml.steps.step_context import StepContext
+
+    sig = inspect.getfullargspec(inspect.unwrap(func))
+    sig_annotations = sig.annotations
+    if "return" in sig_annotations:
+        sig_annotations.pop("return")
+
+    if sig.args and len(sig.args) != len(sig_annotations):
```

Comment:
If `sig.args` enough here or do we also need `sig.kwonlyargs` or something like that?

Reply:
Its enough, I think. See image attached?

![image](https://user-images.githubusercontent.com/4792082/223991139-9a6a71e4-83e9-4a93-abf6-01f9ec12a477.png)